hazelcast/src/main/java/com/hazelcast/cluster/impl/TcpIpJoiner.java

```diff
@@ -45,13 +45,15 @@
 
 public class TcpIpJoiner extends AbstractJoiner {
 
-    private static final int MAX_PORT_TRIES = 3;
     private static final long JOIN_RETRY_WAIT_TIME = 1000L;
 
-    private volatile boolean claimingMaster = false;
+    private final int maxPortTryCount;
+    private volatile boolean claimingMaster;
 
     public TcpIpJoiner(Node node) {
         super(node);
+        int tryCount = node.groupProperties.TCP_JOIN_PORT_TRY_COUNT.getInteger();
```

Comment:
Probably better to throw an exception here (fail fast).

I think we should give the group properties a good cleanup so that its constrains can be defined in the group property itself instead of dealing with it when you get the property.

```
GroupProperty TRY_COUNT = new GroupProperty("com.hazelcast.tcp.tryCount").asInt(..).isPositive(...).build(); 
```